server/modules/surveyRdb/api/surveyRdbApi.js

```diff
@@ -41,17 +52,63 @@ export const init = (app) => {
 
   app.post('/surveyRdb/:surveyId/:nodeDefUuidTable/export', requireRecordListViewPermission, async (req, res, next) => {
     try {
-      const { surveyId, cycle, nodeDefUuidTable } = Request.getParams(req)
-      const nodeDefUuidCols = Request.getJsonParam(req, 'nodeDefUuidCols', [])
-      const filter = Request.getJsonParam(req, 'filter')
-      const sort = Request.getJsonParam(req, 'sort', '')
-
+      const params = {
+        ..._getParamsQuery(req),
+        streamOutput: res,
+      }
       Response.setContentTypeFile(res, 'data.csv', null, Response.contentTypes.csv)
-      const params = { surveyId, cycle, nodeDefUuidTable, nodeDefUuidCols, filter, sort, streamOutput: res }
 
       await SurveyRdbService.fetchViewData(params)
     } catch (error) {
       next(error)
     }
   })
+
+  app.post(
+    '/surveyRdb/:surveyId/:nodeDefUuidTable/query-agg',
+    requireRecordListViewPermission,
+    async (req, res, next) => {
+      try {
+        const params = _getParamsQueryAgg(req)
+        const rows = await SurveyRdbService.fetchViewDataAgg(params)
+
+        res.json(rows)
+      } catch (error) {
+        next(error)
+      }
+    }
+  )
+
+  app.post(
+    '/surveyRdb/:surveyId/:nodeDefUuidTable/query-agg/count',
+    requireRecordListViewPermission,
+    async (req, res, next) => {
+      try {
+        const params = _getParamsQueryAgg(req)
+        const count = await SurveyRdbService.countViewDataAgg(params)
+
+        res.json(count)
+      } catch (error) {
+        next(error)
+      }
+    }
+  )
+
+  app.post(
+    '/surveyRdb/:surveyId/:nodeDefUuidTable/export-agg',
+    requireRecordListViewPermission,
+    async (req, res, next) => {
+      try {
+        const params = {
+          ..._getParamsQueryAgg(req),
+          streamOutput: res,
+        }
+        Response.setContentTypeFile(res, 'data-aggregate.csv', null, Response.contentTypes.csv)
```

Comment:
should we call the files (both raw and aggregate) data-export-${DATE}.csv ?

Reply:
ok